src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -2286,7 +2286,7 @@ private MemberDeclarationSyntax ParseMemberDeclarationOrStatementCore(SyntaxKind
 
                 try
                 {
-                    if ((!haveAttributes || !IsScript) && !haveModifiers && (type.Kind == SyntaxKind.RefType || !IsOperatorKeyword()))
+                    if ((!haveAttributes || !IsScript) && !haveModifiers && (type.Kind == SyntaxKind.RefType || !IsOperatorStart(out _, advanceParser: false)))
```

Comment:
It's not necessary to do as part of this PR or this feature, but I feel like readability would be improved if this condition were commented to indicate that we are checking whether we should try parsing as a top-level statement.